### PR TITLE
Fix a number of pyright typing errors

### PR DESCRIPTION
**Pull request by DominicOram**
### Instructions to reviewer on how to test:
1. Run `pyright` on here and `main` confirm less errors on here
2. Confirm tests still pass

### Checks for reviewer
- [ ] Would the PR title make sense to a scientist on a set of release notes
- [ ] If a new device has been added does it follow the [standards](https://github.com/DiamondLightSource/dodal/wiki/Device-Standards)
- [ ] If changing the API for a pre-existing device, ensure that any beamlines using this device have updated their Bluesky plans accordingly
- [ ] Have the connection tests for the relevant beamline(s) been run via `dodal connect ${BEAMLINE}`

### CODE REVIEW

File: src/dodal/devices/util/epics_util.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def epics_signal_put_wait(pv_name: str, wait: float = 3.0) -> Component[EpicsSig
 
 
 def run_functions_without_blocking(
-    functions_to_chain: list[Callable[[], StatusBase]],
+    functions_to_chain: Sequence[Callable[[], StatusBase]],
```
Comment (rtuck99) [inline src/dodal/devices/util/epics_util.py:29]: :+1: We probably use `list` in far too many places where we could use Sequence

File: tests/devices/unit_tests/test_zebra.py [hunk 2]
```diff
@@ -1,8 +1,7 @@
-from unittest.mock import AsyncMock
+from unittest.mock import AsyncMock, MagicMock
 
 import pytest
 from bluesky.run_engine import RunEngine
-from mockito import mock, verify
```
Comment (rtuck99) [inline tests/devices/unit_tests/test_zebra.py:5]: How many places do we still use mockito? Maybe we can get rid of it as a dependency. I like fewer dependencies.

**Review by rtuck99 — APPROVED** [review]
approved

File: tests/devices/unit_tests/test_zebra.py (hunk 2 above)
Comment (DominicOram) [reply tests/devices/unit_tests/test_zebra.py:5]: Yh, I agree, thought about doing it as part of this but ran out of time. Made https://github.com/DiamondLightSource/dodal/issues/670

File: src/dodal/devices/oav/oav_detector.py [hunk 3]
```diff
@@ -46,6 +46,7 @@ class ZoomController(Device):
     sxst = Component(EpicsSignal, "MP:SELECT.SXST")
 
     def set_flatfield_on_zoom_level_one(self, value):
+        self.parent: "OAV"
```
[the remark this answers is not in the document]
Comment (DominicOram) [reply src/dodal/devices/oav/oav_detector.py:49]: Yh, it's not great. Potentially a sign that the structure here is bad, there's a kind of weird relationship between this device and it's parent which I don't really like. Given it's an `ophyd` device I didn't dwell on it too hard though as it'll be hopefully moved to `ophyd-async` soon anyway

**Pull request merged**